### DISCUSSION TITLE
Pooled connection seems to not be working after idle

### DISCUSSION THREAD

**Discussion by Martichou** [weiznich/diesel_async#202] [Q&A]
Hey! Just starting this discussion because I'm not sure how to diagnose/gather useful logs for this issue.

Lemme explain: I'm using diesel_async with a connection pool (was bb8 previously and now deadpool, but same issue on both). It seems like it I perform a request (through actix, but it's not relevant I think), the first query will succeed as well as any subsequent queries sent immediately after (less than 10s for example).

But then, once I let the server idle for some time (30s-1 minute), the next call will try to use a connection from the pool (I suppose) but that connection will hang until timeout. Then I try another connection, but again, same issue until the pool is fully recreated.

I'm not quite sure how to debug/provide useful information, so if someone know what is needed, I'll do my best!

Thanks

---

ps: note sure if this is related to https://github.com/weiznich/diesel_async/issues/139, but it may be possible. weiznich what's the status on the workaround proposed over there? Still needed/relevant with today version?

**Comment by weiznich** [comment]
I'm sorry, but  I generally do not answer questions if I'm pinged from the question. 
The only exceptions are significant sponsors of diesel and someone that reports a security vulnerability.

**Reply by Martichou** [reply]
My apologies, that was my mistake. My intention wasn’t to demand your immediate attention but rather to clarify that the question was directed to you. I’ll make sure not to ping you like that again. Thank you for your understanding!

**Comment by weiznich** [comment]
It's hard to help you here as you provided almost nothing on information that can help us to reproduce the problem. Ideally you can provide a minimal self contained example that demonstrates the problem. At least the following information are necessary:

* Diesel and Diesel Async version
* Connection pool information (version, which one is used exactly, custom setting?)
* Database system information (which backend is used, which version)
* Operating system version

**Comment by Martichou** [comment]
Ok so after hours of digging, it seems like this has nothing to do with diesel_async nor the Rust code.

I checked traffic using tcpdump and diesel_async (the pool) correctly sends requests and everything to the PG server. But for some reason nothing comes back. Tweaking tcp_keepalive seems to do fix the issue, but not sure if this is related to my OS (Mac), CPE, ... 

Again sorry for the ping. 
Thanks for your answer and the project :)

**Reply by weiznich** [reply]
Thanks for coming back to report that this is resolved. That's always helpful to know.
